provision/cluster/metrics.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cluster
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/tsuru/tsuru/log"
+	"github.com/tsuru/tsuru/servicemanager"
+)
+
+var (
+	desc        = prometheus.NewDesc("tsuru_cluster_info", "Basic information about existing clusters", []string{"provisioner", "name"}, nil)
+	failureDesc = prometheus.NewDesc("tsuru_cluster_fetch_fail", "indicates whether failed to get clusters", []string{}, nil)
+)
+
+func init() {
+	prometheus.MustRegister(&clustersMetricCollector{})
+}
+
+type clustersMetricCollector struct{}
+
+func (c *clustersMetricCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- desc
+}
+
+func (c *clustersMetricCollector) Collect(ch chan<- prometheus.Metric) {
+	clusters, err := servicemanager.Cluster.List()
+	failureValue := float64(0)
+	if err != nil {
+		failureValue = float64(1)
+		log.Errorf("Could not get clusters: %s", err.Error())
+	}
+
+	ch <- prometheus.MustNewConstMetric(failureDesc, prometheus.GaugeValue, failureValue)
```

Comment:
Should not be a counter instead?

Reply:
I choose a gauge because it could be 0, whether `ClusterList` becomes operational after a downtime.